examples/imagenet/alexnet.cc

```diff
@@ -0,0 +1,403 @@
+  dense->set_num_output(num_output);
+
+  ParamSpec *wspec = conf.add_param();
+  wspec->set_name(name + "_weight");
+  wspec->set_decay_mult(wd);
+  auto wfill = wspec->mutable_filler();
+  wfill->set_type("Gaussian");
+  wfill->set_std(std);
+
+  ParamSpec *bspec = conf.add_param();
+  bspec->set_name(name + "_bias");
+  bspec->set_lr_mult(2);
+  bspec->set_decay_mult(0);
+  auto bfill = bspec->mutable_filler();
+  bfill->set_value(bias);
+
+  return conf;
+}
+
+LayerConf GenLRNConf(string name) {
+  LayerConf conf;
+  conf.set_name(name);
+  conf.set_type("CudnnLRN");
+  LRNConf *lrn = conf.mutable_lrn_conf();
+  lrn->set_local_size(5);
+  lrn->set_alpha(1e-04);
+  lrn->set_beta(0.75);
+  return conf;
+}
+
+LayerConf GenFlattenConf(string name) {
+  LayerConf conf;
+  conf.set_name(name);
+  conf.set_type("Flatten");
+  return conf;
+}
+
+LayerConf GenDropoutConf(string name, float dropout_ratio) {
+  LayerConf conf;
+  conf.set_name(name);
+  conf.set_type("CudnnDropout");
+  DropoutConf *dropout = conf.mutable_dropout_conf();
+  dropout->set_dropout_ratio(dropout_ratio);
+  return conf;
+}
+
+FeedForwardNet CreateNet() {
+  FeedForwardNet net;
+  Shape s{3, 227, 227};
+
+  net.Add(new CudnnConvolution(), GenConvConf("conv1", 96, 11, 4, 0, 0.01), &s);
+  net.Add(new CudnnActivation(), GenReLUConf("relu1"));
+  net.Add(new CudnnPooling(), GenPoolingConf("pool1", true, 3, 2, 0));
+  net.Add(new CudnnLRN(), GenLRNConf("lrn1"));
+  net.Add(new CudnnConvolution(),
+          GenConvConf("conv2", 256, 5, 1, 2, 0.01, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu2"));
+  net.Add(new CudnnPooling(), GenPoolingConf("pool2", true, 3, 2, 0));
+  net.Add(new CudnnLRN(), GenLRNConf("lrn2"));
+  net.Add(new CudnnConvolution(), GenConvConf("conv3", 384, 3, 1, 1, 0.01));
+  net.Add(new CudnnActivation(), GenReLUConf("relu3"));
+  net.Add(new CudnnConvolution(),
+          GenConvConf("conv4", 384, 3, 1, 1, 0.01, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu4"));
+  net.Add(new CudnnConvolution(),
+          GenConvConf("conv5", 256, 3, 1, 1, 0.01, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu5"));
+  net.Add(new CudnnPooling(), GenPoolingConf("pool5", true, 3, 2, 0));
+  net.Add(new Flatten(), GenFlattenConf("flat"));
+  net.Add(new Dense(), GenDenseConf("ip6", 4096, 0.005, 1, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu6"));
+  net.Add(new CudnnDropout(), GenDropoutConf("drop6", 0.5));
+  net.Add(new Dense(), GenDenseConf("ip7", 4096, 0.005, 1, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu7"));
+  net.Add(new CudnnDropout(), GenDropoutConf("drop7", 0.5));
+  net.Add(new Dense(), GenDenseConf("ip8", 1000, 0.01, 1));
+
+  return net;
+}
+
+void TrainOneEpoch(FeedForwardNet &net, ILSVRC &data,
+                                      std::shared_ptr<Device> device, int epoch,
+                                      string bin_folder,
+                                      size_t num_train_images,
```

Comment:
would it be better to pass in the total num of training files instead of num of images and images per file?